back/db/migrate/20230321153659_create_permissions_custom_fields.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreatePermissionsCustomFields < ActiveRecord::Migration[6.1]
+  def change
+    create_table :permissions_custom_fields, id: :uuid do |t|
+      t.references :permission, foreign_key: true, type: :uuid, null: false
+      t.references :custom_field, foreign_key: true, type: :uuid, null: false
+      t.boolean :required, null: false, default: false
```

Comment:
Think in the context of permissions required should default to true for custom fields. Probably the more likely scenario in most instances.